### ISSUE TITLE
feat(assimilation): incremental Gauss-Newton 4D-Var wired to the FV forward

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/plumax#93]
## Problem / Request
`solve_4dvar` uses L-BFGS in whitened space. `assimilation/solve.py` ships `run_gauss_newton` (outer linearisation, inner `lineax` CG on `JᵀJ`) and `run_dual_psas` (observation-space, cheap when `dim(y) ≪ dim(x)`), but they are only exercised by the 3D-Var radiance path — the design table lists the incremental solver for Tier III as 🚧. Once the control is a spatial field (large `dim(x)`), the dual PSAS path is exactly the right solver and L-BFGS iteration counts will track `cond(H)`.

## User Story
> As a user with a 100×100-cell control field and a few thousand column observations, I want `solve_4dvar(method="gauss_newton" | "psas")`, so each outer iteration is a structured linear solve rather than hundreds of L-BFGS steps.

## Proposed API
```python
# les_fvm/fourdvar.py
def solve_4dvar(problem, *, method: Literal["lbfgs", "gauss_newton", "psas"] = "lbfgs", outer_iters: int = 5, inner_solver: lineax.AbstractLinearSolver | None = None, ...) -> FourDVarResult: ...
```
Internally: `jax.linearize(problem.forward.predict, x_k)` gives the tangent-linear `H'`; `run_gauss_newton` / `run_dual_psas` take the residual bundle.

## Implementation Steps
- [ ] Expose a residual/Jacobian-vector-product bundle from `FourDVarProblem` in the shape `assimilation.solve` expects
- [ ] Route `method` to `run_lbfgs` / `run_gauss_newton` / `run_dual_psas`; keep `FourDVarResult` (`converged`, `n_steps` from #30)
- [ ] `PosteriorCovariance` reuses the final Gauss-Newton Hessian (already `_gauss_newton_hessian`) instead of rebuilding it
- [ ] Document the "linear forward ⇒ PSAS is exactly the primal solution" equivalence with a test

## Definition of Done
- [ ] On a linear synthetic problem all three methods agree on the MAP to 1e-6 relative
- [ ] Gauss-Newton reaches the same MAP as L-BFGS in ≤ 5 outer iterations on the existing `tests/les_fvm` 4D-Var fixture
- [ ] `make test && make lint && make typecheck` green

## Testing
- [ ] `tests/les_fvm/test_fourdvar_solvers.py::test_methods_agree_linear`
- [ ] `tests/les_fvm/test_fourdvar_solvers.py::test_gauss_newton_outer_iters` (`slow`)

## Relationships
- Parent (theme epic): #72
- Blocked by: spatial per-cell source control vector
- Related: #40 (gaussx quadratic_form in the prior term, closed)
